python/pyspark/errors/__init__.py

```diff
@@ -55,4 +56,5 @@
     "SparkConnectAnalysisException",
     "SparkConnectParseException",
     "SparkConnectTempTableAlreadyExistsException",
+    "SparkConnectIllegalArgumentException",
```

Comment:
I think we can just reuse the existing exception in the future instead of having `SparkConnect` prefix here. We could introduce a new API like `is_remote` in the exception. cc @grundprinzip

Reply:
Yeah, I agree with that because technically they are from the same cause.
Let me refine https://github.com/apache/spark/pull/39783#discussion_r1089869647 and remove the prefix in the follow-ups.